Enhance extension logging documentation (#1434)
microsoft/vscode-react-native: README.md

## Logging

The extension logging is divided by several output channels:

> * React Native - main extension channel which collects outputs from React Native Packager and shows critical errors in the extension

Reviewer: ```suggestion
* React Native - the main extension channel which collects outputs from React Native Packager and shows critical errors in the extension
```

> * These channels spawns only when the specific launch scenario is executed:

Reviewer: ```suggestion
* These channels are spawned only when the specific launch scenario is executed:
```

    * React Native: Run Android
      * LogCat monitor(to get LogCat output from Android device, can be filtered by debug configuration settings)
    * React Native: Run iOS
    * React Native: Run macOS
    * React Native: Run exponent
* Debug Console which is used to receive application logs and logs generated by the debug adapter (`console.log` and other `std` outputs from the app)

> * Extension debugger verbose logs (these logs are shown up if only `trace: "verbose"` option is enabled in debug scenarios)

Reviewer: ```suggestion
* Extension debugger verbose logs (these logs are shown up only if the `trace: "verbose"` option is enabled in debug scenarios)
```

    * React Native Chrome Proxy - shows what runs in and out to the debugger and application

>     * Debug Console become more informative and contain some debugging information from the debug adapter

Reviewer: ```suggestion
    * Debug Console becomes more informative and contains some debugging information from the debug adapter
```

>     * Global extension errors are controlled by VS Code and are printed in VS Code Developer Tools

Reviewer: ```suggestion
    * Global extension errors are controlled by VS Code and printed in VS Code Developer Tools
```
